src/app/utils/socket-utils.ts

```diff
@@ -13,9 +13,9 @@ export function getMasterworkSocketHashes(
   return (masterworkSocketCategory && getPlugHashesFromCategory(masterworkSocketCategory)) || [];
 }
 
-function getPlugHashesFromCategory(category: DimSocketCategory): number[] {
+function getPlugHashesFromCategory(category: DimSocketCategory) {
   return category.sockets
-    .map((socket) => idx(socket, (socket) => socket.plug.plugItem.hash) || NaN)
+    .map((socket) => socket?.plug?.plugItem?.hash || NaN)
```

Comment:
should this use nullish coalescing `??` instead of `||`

Reply:
Yeah I am happy to, I can't imagine we would ever hit `''` or `0` but it is more correct that we look for `null` or `undefined`. I'll fix it up a little later today. On a side note, I didn't even know that was an operator which is cool.

Reply:
Just did a bit of googling on it, is it safe to use yet? The PR to add it to MDN says its only compatible for two browsers so far. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

Reply:
It came with optional chaining in Babel and ts

Reply:
Done